std/math/algebraic.d

```diff
@@ -375,18 +386,23 @@ if (isFloatingPoint!T)
     }
 
     // both are in the normal range
-    return ratio * sqrt(u*u + v*v);
+    return ratio * sqrt(u^^2 + v^^2);
```

Comment:
Unfortunately `^^` is not an inlined operation, so `x^^2` would be slower than `x*x`.

(Possibly this heuristic could be added to the compiler in the `expressionsem.d` for PowExp: `x^^0=>1.0`, `x^^1=>x`, `x^^2=>x*x`)

Reply:
https://github.com/dlang/dmd/issues/21024